Firestore/core/src/firebase/firestore/util/executor.h

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_UTIL_EXECUTOR_H_
+#define FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_UTIL_EXECUTOR_H_
+
+#include <chrono>  // NOLINT(build/c++11)
+#include <functional>
+#include <string>
+#include <utility>
+
+namespace firebase {
+namespace firestore {
+namespace util {
+
+// A handle to an operation scheduled for future execution. The handle may
+// outlive the operation, but it *cannot* outlive the executor that created it.
+class DelayedOperation {
+ public:
+  DelayedOperation() {
+  }
+
+  // If the operation has not been run yet, cancels the operation. Otherwise,
+  // this function is a no-op.
+  void Cancel() {
+    cancel_func_();
+  }
+
+  // Internal use only.
```

Comment:
> Internal use only

Possibly related: If `DelayedOperation` is (eventually) meant to be used by end users *and* this ctor needs to remain public *but* we don't want end users to use it, then we should do the equivalent of `@hide`ing it. The rtdb c++ uses `#if defined(DOXYGEN)` in spots. I suspect you could leverage that to hide this from (eventual) api docs.

OTOH, this isn't in the `include` directory, and I don't think we expose any of these details on the other platforms, so this probably isn't relevant.